plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/SQLEditorBase.java

```diff
@@ -1309,6 +1323,12 @@ protected boolean isNavigationTarget(Annotation annotation) {
         return super.isNavigationTarget(annotation);
     }
 
+    @Nullable
+    @Override
+    public DBPDataSourceContainer getDataSourceContainer() {
+        return null;
```

Comment:
Maybe I missed something, but I don't understand the logic here. You call this method from other methods of this class, however, it always returns null. What's the point of doing that?

Reply:
It's overridden in SQLEditor

Reply:
Logic is still incorrect. Why SQLEditorBase become DBPDataSourceContainerProvider ?
It mustn't return null, at least it should return dataSource.